sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/StateStoreChangelog.scala

```diff
@@ -222,6 +232,31 @@ class StateStoreChangelogWriterV2(
     size += 1
   }
 
+  override def merge(key: Array[Byte], value: Array[Byte]): Unit = {
```

Comment:
This interface is not supported in both v1 and v2 writers. Should we just remove this then ?

Reply:
Thanks for bringing this up. I have a follow up question here - should we also support default column family in changelog v2? Is there a reason to restrict it to only column families?

Reply:
Discussed offline, changelog v2 currently supports both column families, and different record types (enabling support for merge operation). As different record type is required to support merge operations, we will currently make enableColumnFamilies as a prerequisite for merge operation.